homeassistant/components/bluetooth/__init__.py

```diff
@@ -339,19 +341,19 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     slots: int = details.get(ADAPTER_CONNECTION_SLOTS) or DEFAULT_CONNECTION_SLOTS
     entry.async_on_unload(async_register_scanner(hass, scanner, connection_slots=slots))
     await async_update_device(hass, entry, adapter, details)
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = scanner
+    entry.runtime_data = scanner
     entry.async_on_unload(entry.add_update_listener(async_update_listener))
     entry.async_on_unload(scanner.async_stop)
     return True
 
 
-async def async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+async def async_update_listener(
+    hass: HomeAssistant, entry: BluetoothConfigEntry
+) -> None:
     """Handle options update."""
     await hass.config_entries.async_reload(entry.entry_id)
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: BluetoothConfigEntry) -> bool:
     """Unload a config entry."""
-    scanner: HaScanner = hass.data[DOMAIN].pop(entry.entry_id)
-    await scanner.async_stop()
```

Comment:
this was redundant https://github.com/home-assistant/core/pull/117074/files#diff-574d610a162f0473fdbb7e6e91b25d31d42fbb35329c95397b49a58eebd1816dR346

Reply:
Considering that, that means we don't need to store the scanner at all?

Reply:
That is an excellent point 😉

Reply:
96b84cf5c3cd204d066d8bb612027541143da7fa

Reply:
And it means we can revert most of this PR and not store it in `hass.data` at all